test/new-e2e/tests/agent-platform/common/agent_install.go

```diff
@@ -55,6 +56,24 @@ func CheckInstallationMajorAgentVersion(t *testing.T, client *TestClient, expect
 	})
 }
 
+// CheckAgentVersion run tests to check that the agent has the correct version
+func (client *TestClient) CheckAgentVersion(t *testing.T, expected string) bool {
+	return t.Run("Check datadog-agent version", func(t *testing.T) {
+		versionRegexPattern := regexp.MustCompile("^(?m:IoT )?Agent (.*?) -")
+		output := client.AgentClient.Version()
+		matchList := versionRegexPattern.FindStringSubmatch(output)
+		require.Len(t, matchList, 2, "wasn't able to retrieve datadog-agent version on the following output : %s", output)
+
+		// regex to get major.minor.build parts
+		expectedVersion, err := version.New(expected, "")
+		require.NoErrorf(t, err, "invalid expected version %s", expected)
+		actualVersion, err := version.New(matchList[1], "")
+		require.NoErrorf(t, err, "invalid actual version %s", matchList[1])
+
+		require.Equal(t, expectedVersion.GetNumberAndPre(), actualVersion.GetNumberAndPre(), "Expected datadog-agent version %s got %s", expectedVersion, actualVersion)
+	})
+}
```

Comment:
💭: it might be useful to turn the regex matching parts here into a dedicated function that returns just the major, minor, and build of the agent. This is useful and something I suspect we might want to use elsewhere.

Reply:
> This is useful and something I suspect we might want to use elsewhere.

You can use the [version package](https://github.com/DataDog/datadog-agent/blob/main/pkg/version/version.go)